### ISSUE TITLE
License?

### ISSUE THREAD

**Issue by justinkb** [0xf4b1/hearthstone-linux#14]
Just wondering what kind of license you are distributing the stuff in this repo under?

Do you have any objection if I adapt craft.sh into a nix package? This would necessitate publishing it (the nix pkg) under MIT licensing.

**Comment by 0xf4b1** [comment]
I don't mind putting the stuff under MIT and the keg submodule is MIT already. Obviously this will only apply for the files in this repo and I assume you don't plan to package the proprietary game files.

**Comment by justinkb** [comment]
No, I will mark the package with license.unfree, which makes the package
manager aware the package may not be redistributed (it won't be cached by
any continuous integration system that may be involved), so every user who
wants to build the package, has to have the build system download the stuff
via keg themselves.

**Comment by 0xf4b1** [comment]
Okay, that should work out, although I would probably simply remove the building out of the script and package it together with keg to have something like the runtime files or a launcher for the game that is capable of downloading the game and updating it without having to rebuild and reinstall the package again and again for every update.

**Comment by justinkb** [comment]
I won't get into the details of nix, but I have a pretty nix-idiomatic
design in mind. I'm gonna decompose the package into three separate parts.
With how advanced the nix package manager is, with that design, it would be
simpler for users to have grabbing the files integrated. I still have to
make it work, by the way. The unity player uses libraries not declared as
NEEDED in the elf header, so I have to figure out that stuff first. On
nixos you usually need to create a wrapper for such binaries since
libraries aren't installed globally but in an isolated manner. (In fact
/lib doesn't even exist)

**Comment by 0xf4b1** [comment]
Okay, no problem, feel free to report back when you manage to get it working. I've also thought about whether it's worth creating an AUR pkgbuild or snap package, but haven't done it yet.

**Comment by ninetailedtori** [comment]
I'll be making an AUR pkgbuild for this now. Would you like to add an MIT licence to it, officially? I'll leave the license empty so far.

**Comment by ninetailedtori** [comment]
For reference:
```
This is free and unencumbered software released into the public domain.

Anyone is free to copy, modify, publish, use, compile, sell, or distribute this software, either in source code form or as a compiled binary, for any purpose, commercial or non-commercial, and by any means.

In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain. We make this dedication for the benefit of the public at large and to the detriment of our heirs and
successors. We intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.

For more information, please refer to <http://unlicense.org/>
```
This is unlicensed. I'd recommend licensing it with MIT soontm :3

**Comment by ninetailedtori** [comment]
#90
